libcloudforensics/gcp_test.py

```diff
@@ -16,28 +16,343 @@
 
 from __future__ import unicode_literals
 
+import binascii
 import unittest
 
+from googleapiclient.errors import HttpError
+
 from libcloudforensics import gcp
 
 import mock
 import six
 
+FAKE_ANALYSIS_PROJECT = gcp.GoogleCloudProject(
+  'test-target-project',
+  'fake-zone'
+)
+FAKE_ANALYSIS_VM = gcp.GoogleComputeInstance(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'fake-analysis-vm'
+)
+FAKE_INSTANCE = gcp.GoogleComputeInstance(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'fake-instance'
+)
+FAKE_DISK = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'disk1'
+)
+FAKE_BOOT_DISK = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone-other',
+  'boot-disk'
+)
+FAKE_SNAPSHOT = gcp.GoogleComputeSnapshot(
+  FAKE_DISK,
+  'snapshot1'
+)
+FAKE_DISK_COPY = gcp.GoogleComputeDisk(
+  FAKE_ANALYSIS_PROJECT,
+  'fake-zone',
+  'disk1-copy'
+)
+
 
 class GoogleCloudProjectTest(unittest.TestCase):
   """Test Google Cloud project class."""
 
   def setUp(self):
     super(GoogleCloudProjectTest, self).setUp()
-    self.project = gcp.GoogleCloudProject('test-project')
-    self.project.GceApi = mock.MagicMock()
-    self.project.GceOperation = mock.MagicMock()
 
-  def testFormatLogMessage(self):
-    """Test formatting log message."""
+  def test_format_log_message(self):
+    """Test formatting log message. """
     msg = 'Test message'
-    formatted_msg = self.project.format_log_message(msg)
+    formatted_msg = FAKE_ANALYSIS_PROJECT.format_log_message(msg)
     self.assertIsInstance(formatted_msg, six.string_types)
     self.assertEqual(
-        formatted_msg, u'project:{0:s} {1:s}'.format(
-            self.project.project_id, msg))
+      formatted_msg, u'project:{0:s} {1:s}'.format(
```

Comment:
We're python3 only, so the leading `u''` in strings are not necessary

Reply:
Addressed